src/transformers/modeling_utils.py

```diff
@@ -2093,7 +2094,35 @@ def save_pretrained(
 
         # Save the model
         if state_dict is None:
-            state_dict = model_to_save.state_dict()
+            # if model parameters are offloaded, onload and send state dicts to CPU
+            if hasattr(model_to_save, "_hf_hook") and isinstance(model_to_save._hf_hook, AlignDevicesHook):
+                state_dict = {}
+                for name, module in model_to_save.named_modules():
+                    if name == "":
+                        continue
+                    if hasattr(module, "_hf_hook") and isinstance(module._hf_hook, AlignDevicesHook):
```

Comment:
I think that we only need to do this for offloaded modules. You can check if it is a offloaded module through the `module._hf_hook.offload` attribute

Reply:
I think you are right, and have added the check so this only executes if `module._hf_hook.offload` is True